src/metadata/persistence/ontologies.clj

```diff
@@ -1,101 +1,122 @@
+            [next.jdbc.sql :as jsql]
+            [next.jdbc.types :as jtypes]
+            [honey.sql :as sql]
+            [honey.sql.helpers :as h]))
+
+(def ontology-columns [:version :iri :created_by :created_on])
+(defn- ontologies-base-query
+  [cols]
+  (-> (apply h/select cols)
+      (h/from (t "ontologies"))))
 
 (defn add-ontology-xml
   [user version iri ontology-xml]
-  (insert :ontologies (values {:version    version
-                               :iri        iri
-                               :xml        ontology-xml
-                               :created_by user})))
+  (jsql/insert! ds
+                (t "ontologies")
+                {:version    version
+                 :iri        iri
+                 :xml        ontology-xml
+                 :created_by user}))
 
 (defn get-ontology-xml
   [ontology-version]
-  ((comp :xml first)
-   (select :ontologies
-           (fields :xml)
-           (where {:version ontology-version}))))
+  (let [q (-> (ontologies-base-query [:xml])
+              (h/where [:= :version ontology-version]))]
+    (plan/select-one! ds :xml (sql/format q))))
 
 (defn get-ontology-details
   [ontology-version]
-  (first (select :ontologies
-                 (fields :version
-                         :iri
-                         :created_by
-                         :created_on)
-                 (where {:version ontology-version}))))
+  (let [q (-> (ontologies-base-query ontology-columns)
+              (h/where [:= :version ontology-version]))]
+    (plan/select-one! ds ontology-columns (sql/format q))))
 
 (defn set-ontology-deleted
   [ontology-version deleted]
-  (sql/update :ontologies
-              (set-fields {:deleted deleted})
-              (where {:version ontology-version})))
+  (jsql/update! ds
+                (t "ontologies")
+                {:deleted deleted}
+                {:version ontology-version}))
 
 (defn list-ontologies
   []
-  (select :ontologies
-          (fields :version
-                  :iri
-                  :created_by
-                  :created_on)
-          (where {:deleted false})))
+  (let [q (-> (ontologies-base-query ontology-columns)
+              (h/where [:= :deleted false]))]
+    (plan/select! ds ontology-columns (sql/format q))))
+
+(def ontology-class-columns [:iri :label :description])
+(defn- ontology-classes-base-query
+  [cols]
+  (-> (apply h/select cols)
+      (h/from (t "ontology_classes"))))
 
 (defn add-classes
   [ontology-version classes]
   (let [class-values (map #(assoc % :ontology_version ontology-version) classes)]
     (when-not (empty? class-values)
-      (insert :ontology_classes (values class-values)))))
+      (let [new-values (mapv #(vector (:ontology_version %) (:iri %) (:label %) (:description %)) class-values)]
+        (jsql/insert-multi! ds
+                            (t "ontology_classes")
+                            [:ontology_version :iri :label :description]
+                            new-values)))))
```

Comment:
This LGTM as-is, but alternatively, if we wanted to define the set of update columns only once:
```suggestion
      (let [update-cols [:ontology_version :iri :label :description]
            new-values (mapv #(mapv (partial get %) update-cols) class-values)]
        (jsql/insert-multi! ds
                            (t "ontology_classes")
                            update-cols
                            new-values)))))
```
I'm not sure if this way is more confusing, though 🤔

Reply:
Hm, I think I like your version a bit more. I _was_ trying to avoid having the same lists multiple times, but I think I was also getting a bit hurried by the time I was doing ontologies and didn't think of doing it that particular way.